src/java/org/apache/cassandra/security/PEMBasedSslContextFactory.java

```diff
@@ -344,17 +376,40 @@ private void enforceSinglePrivateKeySource()
      */
     private void enforceSingleTurstedCertificatesSource()
     {
-        if (truststoreFileExists() && !StringUtils.isEmpty(pemEncodedCertificates))
+        if (truststoreFileExists() && !StringUtils.isEmpty(pemEncodedTrustCertificates.key))
         {
             throw new IllegalArgumentException("Configuration must specify value for either truststore or " +
                                                "trusted_certificates, not both for PEMBasedSSlContextFactory");
         }
     }
 
+    public static class PEMBasedKeyStoreContext
+    {
+        public String key;
+        public String password;
+        public boolean maybeFilebasedKey;
+
+        public PEMBasedKeyStoreContext(String encodedKey, String getEncodedKeyPassword, boolean maybeFilebasedKey)
```

Comment:
Glad to see that a data class is created for the PEM case too. How about include the `KeyStoreContext`? Therefore, `hasKey()` method look more natural... instead of `hasKey(keyStoreContext.hasKeystore())` in `buildKeyManagerFactory`, the call-site can just have `hasKey()`

```java
        ...
        private final KeyStoreContext fileKeyStore;

        public PEMBasedKeyStoreContext(String encodedKey, String getEncodedKeyPassword, boolean maybeFilebasedKey, KeyStoreContext keyStoreContext)
        {
            this.key = encodedKey;
            this.password = getEncodedKeyPassword;
            this.maybeFilebasedKey = maybeFilebasedKey;
            this.fileKeyStore = keyStoreContext;
        }

        public boolean hasKey()
        {
            return maybeFilebasedKey
                   ? fileKeyStore.hasKeystore()
                   : !StringUtils.isEmpty(key);
        }
```